pkg/storage/internalstorage/storage.go

```diff
@@ -11,18 +13,64 @@ import (
 	"github.com/clusterpedia-io/clusterpedia/pkg/storage"
 )
 
+var mutex sync.Mutex
+
 type StorageFactory struct {
-	db *gorm.DB
+	db             *gorm.DB
+	AutoMigration  *bool
+	DivisionPolicy DivisionPolicy
+	Mapper         []ResourceMapper
+}
+
+func (s *StorageFactory) AutoMigrate() error {
+	return nil
 }
 
 func (s *StorageFactory) GetSupportedRequestVerbs() []string {
 	return []string{"get", "list"}
 }
 
 func (s *StorageFactory) NewResourceStorage(config *storage.ResourceStorageConfig) (storage.ResourceStorage, error) {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	var table string
+	if s.AutoMigration != nil && *s.AutoMigration {
+		switch s.DivisionPolicy {
+		case "", DivisionPolicyNone:
+			table = "resources"
+
+			if exist := s.db.Migrator().HasTable(table); !exist {
+				if err := s.db.AutoMigrate(&Resource{}); err != nil {
+					return nil, err
+				}
+			}
+		case DivisionPolicyGroupResource:
```

Comment:
we do not support DivisionPolicyCustom  in this pr?

Reply:
yes, it not support  it in the pr